control-plane/agents/src/bin/core/volume/operations.rs

```diff
@@ -178,6 +178,19 @@ impl ResourceLifecycle for OperationGuardArc<VolumeSpec> {
             }
         }
 
+        // When nexus is destroyed ahead of the volume destroy, then
+        // delete_nexus_info in previous will not be called since nexus wont be present.
+        // So invoke delete_nexus_info explicitly using the nexus id in target_config if present.
+        if let Some(config) = self.as_ref().config() {
+            let nexus_id = config.target().nexus().clone();
+            // Delete the NexusInfo entry persisted by the IoEngine.
+            ResourceSpecsLocked::delete_nexus_info(
+                &NexusInfoKey::new(&Some(self.uuid().clone()), &nexus_id),
```

Comment:
looks like we take a reference to the nexus_id, so maybe we don't need to clone it on L185?